src/libraries/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.AddSub.cs

```diff
@@ -20,50 +21,50 @@ public static uint[] Add(uint[] left, uint right)
 
             uint[] bits = new uint[left.Length + 1];
 
-            long digit = (long)left[0] + right;
-            bits[0] = unchecked((uint)digit);
+            Add(left, right, ref GetArrayDataReference(bits));
+
+            return bits;
+        }
+
+        private static void Add(ReadOnlySpan<uint> left,
+                                uint right,
+                                ref uint bits)
```

Comment:
Why `ref uint bits` rather than `Span<uint> bits`?

Reply:
Now there is no reason for that because reviewers are OK with bound checks. In the first version I didn't use spans at all, just managed pointers. I'll fix that.